routes/dao_coin_exchange_with_fees.go

```diff
@@ -0,0 +1,1602 @@
+				totalQuantityQuoteCurrencyAfterFeesBaseUnits, err := lib.SafeUint256().Sub(
+					uint256LimitAmount, totalFeeBaseUnits)
+				if err != nil {
+					return nil, fmt.Errorf("HandleMarketOrder: Problem calculating remaining quote currency 1: %v", err)
+				}
+				remainingQuoteQuantityDecimal, err = CalculateStringDecimalAmountFromBaseUnitsSimple(
+					req.QuoteCurrencyPublicKeyBase58Check, totalQuantityQuoteCurrencyAfterFeesBaseUnits)
+				if err != nil {
+					return nil, fmt.Errorf("HandleMarketOrder: Problem calculating remaining quote currency 2: %v", err)
+				}
+			} else {
+				// If there's no price quote then use the simulated amount, minus fees
+				quotCurrencyExecutedAfterFeesBaseUnits, err := lib.SafeUint256().Sub(
+					quoteCurrencyExecutedBeforeFeesBaseUnits, totalFeeBaseUnits)
+				if err != nil {
+					return nil, fmt.Errorf("HandleMarketOrder: Problem calculating remaining quote currency 3: %v", err)
+				}
+				remainingQuoteQuantityDecimal, err = CalculateStringDecimalAmountFromBaseUnitsSimple(
+					req.QuoteCurrencyPublicKeyBase58Check, quotCurrencyExecutedAfterFeesBaseUnits)
+				if err != nil {
+					return nil, fmt.Errorf("HandleMarketOrder: Problem calculating remaining quote currency 4: %v", err)
+				}
+			}
+		} else {
+			// Just to be safe, catch an error here
+			return nil, fmt.Errorf("HandleMarketOrder: Invalid quantity currency type %v",
+				req.QuantityCurrencyType)
+		}
+		if remainingQuoteQuantityDecimal == "" {
+			return nil, fmt.Errorf("HandleMarketOrder: Problem calculating remaining quote currency 5")
+		}
+
+		// Now we need to execute the order with the remaining quote currency.
+		// To make this simple and exact, we can do this as an ask where we are
+		// selling the quote currency for base currency. This allows us to specify
+		// the amount of quote currency as the quantity. To make this work we must
+		// also set the price to the inverse of the quote price because ask orders
+		// specify their price in (buying coin per selling coin), which in this case
+		// is (base / quote), or the inversion of priceQuoteStr. Again consensus is
+		// confusing sorry about that...
+		priceStrQuoteInverted, err := InvertPriceStr(priceStrQuote)
+		if err != nil {
+			return nil, fmt.Errorf("HandleMarketOrder: Problem inverting price: %v", err)
+		}
+		newDaoCoinMarketOrderRequest := &DAOCoinLimitOrderCreationRequest{
+			TransactorPublicKeyBase58Check:            req.TransactorPublicKeyBase58Check,
+			BuyingDAOCoinCreatorPublicKeyBase58Check:  req.BaseCurrencyPublicKeyBase58Check,
+			SellingDAOCoinCreatorPublicKeyBase58Check: req.QuoteCurrencyPublicKeyBase58Check,
+			Quantity:             remainingQuoteQuantityDecimal,
+			OperationType:        DAOCoinLimitOrderOperationTypeStringASK,
+			Price:                priceStrQuoteInverted,
+			FillType:             req.FillType,
+			MinFeeRateNanosPerKB: req.MinFeeRateNanosPerKB,
+			TransactionFees:      req.TransactionFees,
```

Comment:
should merge in the node-level configured transaction fees.

Reply:
I fixed this.